### ISSUE TITLE
Sort by Owned in Collection View does not work

### ISSUE THREAD

**Issue by mattsigal** [fafranco82/swdestinydb#45]
Tried to sort my collection by the "Owned" column (so the cards that I have most of would filter to the top or bottom) but it doesn't trigger.  This makes sense for complex columns (like Die), but I feel like Owned should be sortable.

**Comment by fafranco82** [comment]
Well, actually the "owned" column is a complex object like the die (it has two values... one for cards and another one for dice). Anyway, I will implement some way to sort them properly.

**Comment by mattsigal** [comment]
Yeah, I realized that after I posted, haha. Since so many cards do not have dice, I think it would make sense for Owned to just sort by the first column (# of cards, toggling ascending/descending), and not even bother adding sort by # of dice.

**Comment by stopfstedt** [comment]
the number of cards and dice owned are stored in a nested object property `owned` of those cards.
so i tried to fix this by setting `data-sort="owned.cards"` on that "Owned" link in the collections-table header.
doesn't seem to work, tho.

**Comment by stopfstedt** [comment]
a possible solution would be to deconstruct the `owned` object and replace it with `owned_cards` and `owned_dice` attributes on the card objects. 
will take a stab at this.

**Comment by fafranco82** [comment]
Maybe in the collection page, and deconstructing without removing original... but you must update also the new fields when collection changes.

The problem is Forerunnerdb doesn't interpret dot notation, but I think using a lodash function I can transform `owned.cards` to `{owned: cards}`, that I think Forerunnedb does interpret right. I will try.

**Comment by fafranco82** [comment]
I know what is happening. `owned.cards` is a valid key for sorting... but in the code, if there is no `name` key for sorting, it is added (I supposed for untying), but, as `$orderBy` object is not a list, take precedence `name` sorting (or maybe because single property take precedence over nested properties). I need to investigate more about this, but I'm really starved, so I'm going to have a dinner and then sleep :)

**Comment by fafranco82** [comment]
Fixed with 9f61e108c8a8d60f4c421cd570033e24e962e831. Updating ForerunnerDB...

**Comment by fafranco82** [comment]
Unfortunately, I didn't test it well... it broke decklists, so I revert the library :(

**Comment by ogreboy2** [comment]
I don't have any programming skills, alas, but I'm happy to offer a bounty on this bug in the hopes that it gets fixed faster. :-) It drives me nuts that I can't properly sort by all the column headings, especially this one.
